src/openrct2/entity/EntityRegistry.cpp

```diff
@@ -275,7 +275,14 @@ static void AddToEntityList(EntityBase* entity)
 {
     auto& list = gEntityLists[EnumValue(entity->Type)];
     // Entity list must be in sprite_index order to prevent desync issues
-    list.insert(std::lower_bound(std::begin(list), std::end(list), entity->Id), entity->Id);
+    if (!list.empty() && list.back() < entity->Id)
```

Comment:
if the list is empty this also works so could do `list.empty() || list.back() < entity->id` (but since this is only going to ever happen once per save its not like it makes any performance difference)